src/service/extension-location.js

```diff
@@ -68,14 +92,19 @@ export function calculateExtensionScriptUrl(
   opt_extensionVersion,
   opt_isLocalDev
 ) {
-  const base = calculateScriptBaseUrl(location, opt_isLocalDev);
+  const useDefaultCdn =
+    urls.useDefaultCdnForDynamicComponents && isDynamicComponent(extensionId);
+  const base = calculateScriptBaseUrl(location, opt_isLocalDev, useDefaultCdn);
   const rtv = getMode().rtvVersion;
   if (opt_extensionVersion == null) {
     opt_extensionVersion = '0.1';
   }
   const extensionVersion = opt_extensionVersion
     ? '-' + opt_extensionVersion
     : '';
+  if (urls.cdn !== urls.defaultCdn && !useDefaultCdn) {
```

Comment:
If single pass is enabled does this logic still function as expected? For both conditions.

Reply:
Ah, I mistakenly assumed single pass mode was related to localdev, not experiments. I will rewrite this to include `spPath`.

Reply:
This is resolved in commit d76da0c. Let me know how the changes look to you.

Reply:
if i'm understanding it correctly https://github.com/ampproject/amphtml/commit/d76da0c49090586b6048645fe948c5005b983629 **could** create a path that looks like `cdn.ampproject.org/rtv/${rtvVersion}/sp/v0/${extensionId}${extensionVersion}.js`? if so that might break. 

We can actually remove that `getSinglePassExperimentPath()` logic now since we assign unique versions to the resources we deploy as a single pass experiment.

feel free to ping me if you need further help on this part! (or ping me on slack too if you want a quicker response time `erwinm`)

Reply:
just to make my explanation simpler, the whole `sp` infix logic is moot if its served on google amp cache (even with single pass experiment ON). it was needed in an old version of our experiment deployment but no longer needed.

Reply:
@erwinmombay Since removal of single pass experiment paths is unrelated to this PR topic, would you mind handling that change in a separate commit?

Reply:
@mattwomple will do. I'll take it on today. 👍

Reply:
changes merged here https://github.com/ampproject/amphtml/pull/25073